src/models/StyledNativeComponent.js

```diff
@@ -95,8 +96,23 @@ export default (constructWithOptions: Function) => {
       }
     }
 
+    setNativeProps(nativeProps: Object) {
+      this.root.setNativeProps(nativeProps)
+    }
+
+    generateRef() {
+      const { innerRef } = this.props
+      return (component: any) => {
+        this.root = component
+
+        if (innerRef) {
```

Comment:
This is hurting perf on a micro level and since we're having efforts to improve perf at the same time it'd be great to fix this and just swap it out with a simple `typeof innerRef === 'function'` please